src/components/Tooltip/index.tsx

```diff
@@ -1,147 +1,262 @@
-import { Theme, css } from '@emotion/react'
 import styled from '@emotion/styled'
-import PropTypes from 'prop-types'
-import { ReactNode, useEffect } from 'react'
 import {
-  Tooltip as ReakitTooltip,
-  TooltipArrow,
-  TooltipReference,
-  useTooltipState,
-} from 'reakit/Tooltip'
-import Box, { BoxProps } from '../Box'
-
-type TooltipPlacement =
-  | 'auto-start'
-  | 'auto'
-  | 'auto-end'
-  | 'top-start'
-  | 'top'
-  | 'top-end'
-  | 'right-start'
-  | 'right'
-  | 'right-end'
-  | 'bottom-end'
-  | 'bottom'
-  | 'bottom-start'
-  | 'left-end'
-  | 'left'
-  | 'left-start'
-
-const variants = {
-  black: ({ theme }: { theme: Theme }) => css`
-    background-color: ${theme.colors.neutral.backgroundStronger};
-    color: ${theme.colors.neutral.textStronger};
-    fill: ${theme.colors.neutral.backgroundStronger};
-    box-shadow: ${theme.shadows.tooltip};
-  `,
-  white: ({ theme }: { theme: Theme }) => css`
-    background-color: ${theme.colors.neutral.backgroundWeak};
-    color: ${theme.colors.neutral.text};
-    fill: ${theme.colors.neutral.backgroundWeak};
-    box-shadow: ${theme.shadows.tooltip};
-  `,
-}
+  ReactNode,
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import { createPortal } from 'react-dom'
+import { getUUID } from '../../utils'
```

Comment:
can we use useId from react now ?

Reply:
We are not yet using react 18 here

Reply:
It will be replaced when we will be using react 18 :)